### PR TITLE
Adds checking for trailling commas on the project

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

This adds checking for trailing commas on our linter.

This should even adds checking on the Editors
Captura de Tela 2021-02-17 às 14 52 50

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [ ] This branch is based on the latest `master`
- [ ] I have added an entry under `[next]` in `CHANGELOG.md`
- [ ] I have formatted my code with `flutter format`
- [ ] I have made corresponding changes to the documentation
- [ ] I have added examples for new features in `doc/examples`
- [ ] The continuous integration (CI) is passing

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -72,3 +75,12 @@ linter:
     - use_rethrow_when_possible
     - unnecessary_new
 
+dart_code_metrics:
+  rules:
+    - prefer-trailing-comma
+  metrics:
```
Comment (erickzanardo) [inline analysis_options.yaml:81]: Note to reviewers: I couldn't find a way for the lib to not check those, so I set the rules accordingly to our current scenario.

File: lib/src/components/joystick/joystick_events.dart [hunk 2]
```diff
@@ -7,7 +7,7 @@ enum JoystickMoveDirectional {
   MOVE_DOWN_RIGHT,
   MOVE_DOWN_LEFT,
   MOVE_LEFT,
-  IDLE
+  IDLE,
```
Comment (erickzanardo) [inline lib/src/components/joystick/joystick_events.dart:10]: Note to reviewers: This line broke on my editor, but not on the linter, could someone open this file to confirm if this is a bug on my plugin?

Comment (luanpotter) [reply lib/src/components/joystick/joystick_events.dart:10]: working fine for me. and I believe dart supports trailing commas on enums so should work?

**Review by luanpotter — APPROVED** [review]
YEEEEEEEES! That is marvelous @erickzanardo !

File: lib/src/components/joystick/joystick_events.dart (hunk 2 above)
Comment (erickzanardo) [reply lib/src/components/joystick/joystick_events.dart:10]: Nice. I imagined it was a bug on my plugin. But it vanished eventually, thanks for confirming

**Review by spydon — APPROVED** [review]

**Pull request merged**